src/main/java/com/liorkn/elasticsearch/script/VectorScoreScript.java

```diff
@@ -1,114 +1,90 @@
-
-storing arrays is no luck - lucine index doesn't keep the array members orders
-https://www.elastic.co/guide/en/elasticsearch/guide/current/complex-core-fields.html
-
-Delimited Payload Token Filter: https://www.elastic.co/guide/en/elasticsearch/reference/2.4/analysis-delimited-payload-tokenfilter.html
-
-
- */
-
 package com.liorkn.elasticsearch.script;
 
-import com.liorkn.elasticsearch.Util;
-import org.apache.lucene.index.BinaryDocValues;
-import org.apache.lucene.store.ByteArrayDataInput;
-import org.elasticsearch.common.Nullable;
-import org.elasticsearch.script.ExecutableScript;
-import org.elasticsearch.script.LeafSearchScript;
-import org.elasticsearch.script.ScriptException;
-
-import java.nio.ByteBuffer;
-import java.nio.DoubleBuffer;
+import java.io.IOException;
+import java.io.UncheckedIOException;
 import java.util.ArrayList;
-import java.util.Base64;
 import java.util.Map;
 
-/**
- * Script that scores documents based on cosine similarity embedding vectors.
- */
-public final class VectorScoreScript implements LeafSearchScript, ExecutableScript {
-
-    public static final String SCRIPT_NAME = "binary_vector_score";
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.store.ByteArrayDataInput;
+import org.elasticsearch.script.SearchScript;
+import org.elasticsearch.search.lookup.SearchLookup;
 
-    private static final int DOUBLE_SIZE = 8;
+import com.liorkn.elasticsearch.Util;
 
-    // the field containing the vectors to be scored against
-    public final String field;
+public final class VectorScoreScript extends SearchScript {
 
-    private int docId;
     private BinaryDocValues binaryEmbeddingReader;
-
-    private final double[] inputVector;
-    private final double magnitude;
-
+    
+	private final String field;
     private final boolean cosine;
 
+    private final float[] inputVector;
+    private final float magnitude;
+    
     @Override
     public long runAsLong() {
-        return ((Number)this.run()).longValue();
-    }
-    @Override
-    public double runAsDouble() {
-        return ((Number)this.run()).doubleValue();
-    }
-    @Override
-    public void setNextVar(String name, Object value) {}
-    @Override
-    public void setDocument(int docId) {
-        this.docId = docId;
+        return (long) runAsDouble();
     }
+    
+	@Override
+	public double runAsDouble() {
+		try {
+            final byte[] bytes = binaryEmbeddingReader.binaryValue().bytes;
+            final ByteArrayDataInput input = new ByteArrayDataInput(bytes);
+            
+            input.readVInt(); // returns the number of values which should be 1, MUST appear hear since it affect the next calls
+            
+            final int len = input.readVInt();
+            // in case vector is of different size
+            if (len != inputVector.length * Double.BYTES) {
```

Comment:
this will always be true since we moved to floats.
I would like the scoring to fail if we have a problem with the vectors instead of hiding it by returning 0.
It helped us find bugs in our vector creation code. So please remove this check & the try..catch on the entire function

Reply:
I cannot remove try/catch - binaryEmbeddingReader.binaryValue() can throw an IOException:
    public abstract BytesRef binaryValue() throws IOException